### ISSUE TITLE
Missing bacio module load - build fails on WCOSS2

### ISSUE THREAD

**Issue by KateFriedman-NOAA** [NOAA-EMC/GSI-Monitor#12]
The GSI-Monitor build fails on WCOSS2 with an error message about `bacio` (using `w3emc/2.9.2`):

```
-- Found ncdiag: /apps/ops/para/libs/intel/19.1.3.304/cray-mpich/8.1.7/ncdiag/1.0.0/lib64/libncdiag.a (found version "1.0.0")
CMake Error at /apps/spack/cmake/3.20.2/intel/19.1.3.304/utnbptm3hrf7gppztidueu4jogfgemut/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findbacio.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "bacio", but
  CMake did not find one.

  Could not find a package configuration file provided by "bacio" with any of
  the following names:

    bacioConfig.cmake
    bacio-config.cmake

  Add the installation prefix of "bacio" to CMAKE_PREFIX_PATH or set
  "bacio_DIR" to a directory containing one of the above files.  If "bacio"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /apps/ops/para/libs/intel/19.1.3.304/w3emc/2.9.2/lib/cmake/w3emc/w3emc-config.cmake:43 (find_dependency)
  CMakeLists.txt:48 (find_package)


-- Configuring incomplete, errors occurred!
See also "/lfs/h2/emc/global/noscrub/kate.friedman/git/fork_feature_dev-wcoss2/sorc/gsi_monitor.fd/build/CMakeFiles/CMakeOutput.log".
```

Neither the `wcoss2.lua` or `common.lua` modulefiles load the `bacio` module. I added `bacio/2.4.1` to `common.lua` and the build worked on WCOSS2 (Dogwood). The default `w3emc` version is 2.9.1 but I believe we need to use 2.9.2 on WCOSS2 now.

Change I made to `common.lua`:

```
diff --git a/modulefiles/common.lua b/modulefiles/common.lua
index b26b234..a7f5563 100644
--- a/modulefiles/common.lua
+++ b/modulefiles/common.lua
@@ -3,9 +3,11 @@ Load common modules to build GSI Monitoring on all machines
 ]])
 
 local netcdf_ver=os.getenv("netcdf_ver") or "4.7.4"
+local bacio=os.getenv("bacio_ver") or "2.4.1"
 local w3emc_ver=os.getenv("w3emc_ver") or "2.9.1"
 local ncdiag_ver=os.getenv("ncdiag_ver") or "1.0.0"
 
 load(pathJoin("netcdf", netcdf_ver))
+load(pathJoin("bacio", bacio_ver))
 load(pathJoin("w3emc", w3emc_ver))
 load(pathJoin("ncdiag", ncdiag_ver))
```

Build logs (Dogwood, can repeat on Cactus when it's back):
failed: /lfs/h2/emc/global/noscrub/kate.friedman/git/fork_feature_dev-wcoss2/sorc/logs/build_gsi_monitor.log_test
worked: /lfs/h2/emc/global/noscrub/kate.friedman/git/fork_feature_dev-wcoss2/sorc/logs/build_gsi_monitor.log_test2

I did not add `bacio` and test the build on other platforms.

**Comment by aerorahul** [comment]
This one is on me @EdwardSafford-NOAA 
The CI passed because w3emc/2.9.2 automatically fetches its dependencies in spack.
The modulefiles however loads w3emc/2.9.1. w3emc/2.9.1 has a bug in its cmake config, in that it does not list bacio as a dependency.  As a result we were able to build on orion and hera.

To fix this issue:
- load bacio/2.4.1 in `common.lua` and it should fix the issue.

Also, now that WCOSS Dell is gone, we can remove those modulefiles.

**Comment by EdwardSafford-NOAA** [comment]
Closed with PR #13.
